File: frontend/src/objects/rider/profile/RiderPayout.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type { CurrencyCents } from '@/objects/core/SharedObjects'
import type { MerchantPayoutAccount } from '@/objects/merchant/profile/MerchantPayoutAccount'
import type { MerchantWithdrawal } from '@/objects/merchant/profile/MerchantWithdrawal'

export type RiderPayout = {
  payoutAccount?: MerchantPayoutAccount
  withdrawnCents: CurrencyCents
  availableToWithdrawCents: CurrencyCents
  withdrawalHistory: MerchantWithdrawal[]
}

File: backend/src/main/scala/services/rider/objects/RiderPayout.scala
package services.rider.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import services.merchant.objects.{MerchantPayoutAccount, MerchantWithdrawal}
import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class RiderPayout(
    payoutAccount: Option[MerchantPayoutAccount],
    withdrawnCents: CurrencyCents,
    availableToWithdrawCents: CurrencyCents,
    withdrawalHistory: List[MerchantWithdrawal],
)
object RiderPayout:
  given Encoder[RiderPayout] = deriveEncoder
  given Decoder[RiderPayout] = deriveDecoder
